asana/asana.go

```diff
@@ -189,8 +172,35 @@ type (
 
 	// Errors always has at least 1 element when returned.
 	Errors []Error
+
+	// HTTP request error
+	requestError struct {
+		Body string
+		Code int
```

Comment:
I have a problem here where if I run the tests and reference `err.Code` I get this error: `err.Code undefined (type error has no field or method Code)`. I'd like to avoid the caller to having to cast `err` as a different type to get at the `Code` value—do you know of a better pattern here?

Reply:
There's two options. One is to export the struct and require a cast, this is pretty typical for error types.

The second, would be to expose a `HTTPStatus() int` method and then use a private interface to check the type. In our server code you'll see we often use `Code() codes.Code` which returns aligns with the grpc error types.

Reply:
Cool, just updated. I left it at casting since it's normal.